packages/contracts/contracts/L1/L2OutputOracle.sol

```diff
@@ -0,0 +1,120 @@
+//SPDX-License-Identifier: MIT
+pragma solidity >=0.8.10;
+
+import { Ownable } from "../../lib/openzeppelin-contracts/contracts/access/Ownable.sol";
+
+/**
+ * @title L2OutputOracle
+ */
+// The payable keyword is used on appendL2Output to save gas on the msg.value check.
+// slither-disable-next-line locked-ether
+contract L2OutputOracle is Ownable {
+    event l2OutputAppended(bytes32 indexed _l2Output, uint256 indexed _l2timestamp);
+
+    // The interval in seconds at which checkpoints must be submitted.
+    uint256 public immutable submissionInterval;
+
+    // The time between blocks on L2.
+    uint256 public immutable l2BlockTime;
+
+    // The number of blocks in the chain before the first block in this contract.
+    uint256 public immutable historicalTotalBlocks;
+
+    // The timestamp of the first L2 block recorded in this contract.
+    uint256 public immutable startingBlockTimestamp;
+
+    // The timestamp of the most recent L2 block recorded in this contract.
+    uint256 public latestBlockTimestamp;
+
+    // A mapping from L2 timestamps to the output root for the block with that timestamp
+    mapping(uint256 => bytes32) public l2Outputs;
```

Comment:
Any opinions on making this internal and adding a getter function? I'm not super opinionated